### PR TITLE
chore: update wasmtime to v24

### PR DESCRIPTION
**Pull request by Stebalien**

### CODE REVIEW

**Comment by Stebalien** [comment]
Ok, this appears to work: https://github.com/filecoin-project/lotus/pull/12449

File: fvm/src/engine/mod.rs [hunk 1]
```diff
@@ -164,6 +166,7 @@ fn wasmtime_config(ec: &EngineConfig) -> anyhow::Result<wasmtime::Config> {
 
     // wasmtime default: true
     // We don't want threads, there is no way to ensure determinism
+    #[cfg(feature = "wasmtime/threads")]
     c.wasm_threads(false);
```
Comment (rvagg) [inline fvm/src/engine/mod.rs:170]: should we just remove the feature then? or are you wanting to keep this just to be explicit about it for if it eventually shows up not behind a feature flag?

Comment (rvagg) [reply fvm/src/engine/mod.rs:170]: ah, it's on by default anyway so always tehre

**Comment by rvagg** [comment]
line 180 of fvm/src/engine/mod.rs, `wasm_tail_call` is default is now `true`, just needs a doc update

**Review by rvagg — APPROVED** [review]

**Pull request merged**